app/uk/gov/hmrc/exports/models/Pointer.scala

```diff
@@ -25,19 +25,30 @@ import scala.util.{Success, Try}
 object PointerSectionType extends Enumeration {
   type PointerSectionType = Value
   val FIELD, SEQUENCE = Value
-  implicit val format: Format[models.PointerSectionType.Value] =
-    Format(Reads.enumNameReads(PointerSectionType), Writes.enumNameWrites)
+  implicit val format: Format[models.PointerSectionType.Value] = Format(Reads.enumNameReads(PointerSectionType), Writes.enumNameWrites)
 }
 
 case class PointerSection(value: String, `type`: PointerSectionType) {
   lazy val pattern: String = `type` match {
     case PointerSectionType.FIELD    => value
     case PointerSectionType.SEQUENCE => "$"
   }
+
+  override def toString: String = `type` match {
+    case PointerSectionType.FIELD    => value
+    case PointerSectionType.SEQUENCE => "#" + value
+  }
 }
 
 object PointerSection {
-  implicit val format = Json.format[PointerSection]
+  private val SEQUENCE_REGEX = "^#(\\d*)$".r
+  implicit val format: Format[PointerSection] =
+    Format[PointerSection](Reads(js => js.validate[String].map(PointerSection(_))), Writes(section => JsString(section.toString)))
+
+  def apply(value: String): PointerSection = SEQUENCE_REGEX.findFirstMatchIn(value).map(_.group(1)) match {
```

Comment:
What this 1 means? 
`The matched string in group i, or null if nothing was matched.`

This is from documentation, do we need this `group(1)`?

What do you thinki about using `matched`?
https://www.scala-lang.org/api/2.12.1/scala/util/matching/Regex$$Match.html#matched:String

Reply:
Tbh a Regex was probably a bit overkill for this... All It is doing is checking if the string starts with a `#` and then it extracts the number after it if it does

Reply:
Could easily be `.filter(_.startsWith("#")).map(_.replaceAll("#", ""))` I guess

Reply:
Or something along those lines